lib/spack/spack/compilers/__init__.py

```diff
@@ -255,41 +255,69 @@ def compilers_for_arch(arch_spec, scope=None):
     return list(get_compilers(config, arch_spec=arch_spec))
 
 
-def compiler_from_config_entry(items):
-    config_id = id(items)
-    compiler = _compiler_cache.get(config_id, None)
+class StrongReference(object):
```

Comment:
Do you envision this class as having more uses than as a (compiler) cache key?  If not, I think a more specific name is more appropriate.

Reply:
Generally I think this class will have limited use, but I was wary of giving it a more specific name to avoid the implication that it did something specific with the compiler objects. It might belong in a utility module rather than in the `compilers` module.

Reply:
I like the idea of this being a utility, especially if it might be re-used in another context.

What about calling the class something like `CacheReference`?  That could provide a bit more context to the purpose for such a class.